src/Core/tree_policies/UCT_TreePolicy.cpp

```diff
@@ -0,0 +1,17 @@
+#include <UCT_TreePolicy.h>
+
+SearchNode UCT_TreePolicy::treePolicy(SearchNode &node) {
+
+    SearchNode c_node = node;
+    while (c_node.isTerminalState) {
+
+        if (c_node.unvisited_child_states.size() != 0) {
+            SearchNode expanded_node = m_expand(c_node);
+            return expanded_node;
+        }
+
+        SearchNode best_child = m_bestChild(node);
```

Comment:
I think this should be m_bestChild(c_node)?
Also, is it necessary to do this in 2 steps?